src/main/java/com/wolox/servingwebcontent/BookController.java

```diff
@@ -1,16 +1,72 @@
 package com.wolox.servingwebcontent;
-
-import org.springframework.stereotype.Controller;
+import com.wolox.servingwebcontent.exceptions.BookIdMismatchException;
+import com.wolox.servingwebcontent.exceptions.BookNotFoundException;
+import com.wolox.servingwebcontent.models.Book;
+import com.wolox.servingwebcontent.packages.BookRepository;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/title/{bookTitle}")
+    public Optional<Book> findByTitle(@PathVariable String bookTitle) {
+        return bookRepository.findByTitle(bookTitle);
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id) {
+        return bookRepository.findById(id)
+            .orElseThrow(() -> new BookNotFoundException("Book not found"));
```

Comment:
If you make the correction in the exception you'll be able to do `.orElseThrow(BookNotFoundException::new)`